app.py

```diff
@@ -263,24 +219,11 @@ def evaluate_policy():
     action = app.current_request.json_body["action"]
     resource = app.current_request.json_body["resource"]
     try:
-        provision_user(principal)
+        user = User(directory, principal)
     except Exception:
         pass
-
-    username = quote(principal)
-    policies_to_evaluate = []
-    for ppl in ad.lookup_policy(DirectoryArn=directory_arn,
-                                ObjectReference=dict(Selector="/" + username))["PolicyToPathList"]:
-        for policy_info in ppl["Policies"]:
-            if "PolicyId" in policy_info:
-                print(policy_info["PolicyId"])
-                for attr in ad.list_object_attributes(DirectoryArn=directory_arn,
-                                                      ObjectReference=dict(
-                                                          Selector="$" + policy_info["PolicyId"]))["Attributes"]:
-                    if attr["Key"]["Name"] == "policy_document":
-                        policies_to_evaluate.append(attr["Value"]["BinaryValue"].decode())
-
-    result = iam.simulate_custom_policy(PolicyInputList=policies_to_evaluate,
+    user.lookup_policies()
```

Comment:
If the above is a `pass`, then `user` won't be declared and your error is going to be pretty confusing.

Reply:
fixed